packages/misc/src/lib/helper/session-sigs-validator.ts

```diff
@@ -0,0 +1,208 @@
+import {
+  AuthSig,
+  Capability,
+  ParsedSessionMessage,
+  ParsedSignedMessage,
+  SessionSigsMap,
+} from '@lit-protocol/types';
+
+interface ValidationResult {
+  isValid: boolean;
+  errors: string[];
+}
+
+// Function to parse a signedMessage string into an object
+function parseSignedMessage(signedMessage: string): ParsedSignedMessage {
+  const lines = signedMessage.split('\n');
+  const parsedData: ParsedSignedMessage = {};
+  let currentKey: string | null = null as string | null;
+  let currentValue = '';
+
+  lines.forEach((line) => {
```

Comment:
Are we sure that `currentValue` being assigned from within a `forEach` loop will not result in `currentKey` being a value other than the last key value pair? since iteration of the loop is not guaranteed by the time we wish to process the value?

Reply:
each line is processed one by one, in order, without skipping, so I don't understand why would it be overlapping?

Reply:
since `forEach` is actually sync and does not return a promise but `undefined` this is actually fine.